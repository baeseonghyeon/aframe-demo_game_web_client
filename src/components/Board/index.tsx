import React, { Children } from 'react';
import styles from './Board.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

interface BoardProps {
  children?: React.ReactNode;
  title?: string;
  className?: string;
}

const Board = (props: BoardProps) => {
  const { children, title = '', className } = props;

  return (
    <div className={cn('container', className)}>
      {title && (
        <div className={cn('title__container')}>
          <h1 className={cn('title')}>{title}</h1>
        </div>
      )}
      <div className={cn('wrapper')}>{children}</div>
    </div>
  );
};

export default Board;
