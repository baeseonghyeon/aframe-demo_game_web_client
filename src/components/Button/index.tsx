import React from 'react';
import styles from './Button.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

interface ButtonProps {
  children?: React.ReactNode;
  className?: string;
  /**
   * 버튼 테마
   */
  theme?: 'pink' | 'yellow';
  /**
   * 버튼 곡선
   */
  shape?: 'circle';
}

const Button = (props: ButtonProps) => {
  const { theme = 'pink', shape, children, className } = props;

  return (
    <button
      className={cn(
        'container',
        `theme--${theme}`,
        `shape--${shape}`,
        className
      )}
    >
      {children}
    </button>
  );
};

export default Button;
