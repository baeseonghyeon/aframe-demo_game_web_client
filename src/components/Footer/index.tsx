import React from 'react';
import styles from './Footer.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

const Footer = () => {
  return <div className={cn('container')}>Â© 2021 A-Frame Demo Game</div>;
};

export default Footer;
