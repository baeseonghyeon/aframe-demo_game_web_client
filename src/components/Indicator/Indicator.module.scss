@import '../../scss/global';

// ==============================================
// Cicle Indicator
// ==============================================

.indicator {
  width: 100px;
  height: 100px;
  position: absolute;
  top: calc(50% - 50px);
  left: calc(50% - 50px);
  animation: spin 1s linear infinite;
  z-index: 1;
}

.circle {
  fill: none;
  // stroke: rgba(255, 255, 255, 0.5);
  stroke: $color-primary;
  stroke-linecap: round;
  stroke-width: 4;
  animation: draw 3s ease-in-out infinite;
}

@keyframes draw {
  0% {
    stroke-dasharray: 20, 282.6;
  }
  50% {
    stroke-dasharray: 200, 282.6;
  }
  100% {
    stroke-dasharray: 20, 282.6;
  }
}

@keyframes spin {
  to {
    transform: rotate(360deg);
  }
}

// ==============================================
// Mixins
// ==============================================

@mixin dot(
  $width: $dot-width,
  $height: $dot-height,
  $radius: $dot-radius,
  $bg-color: $dot-bg-color,
  $color: $dot-color
) {
  width: $width;
  height: $height;
  border-radius: $radius;
  background-color: $bg-color;
  color: $color;
}

// ==============================================
// Variables
// ==============================================

$dot-width: 15px !default;
$dot-height: 15px !default;
$dot-radius: $dot-width/2 !default;

$dot-color: rgba($color-primary, 0.9) !default;
$dot-bg-color: $dot-color !default;
$dot-before-color: $dot-color !default;
$dot-after-color: $dot-color !default;

$dot-spacing: $dot-width * 2.5 !default;

/**
 * ==============================================
 * Dot Flashing
 * ==============================================
 */

.dot-flashing {
  position: relative;
  z-index: 1;
  @include dot;

  animation: dot-flashing 1s infinite linear alternate;
  animation-delay: 0.5s;

  &::before,
  &::after {
    content: '';
    display: inline-block;
    position: absolute;
    top: 0;
  }

  &::before {
    left: -$dot-spacing;

    @include dot($bg-color: $dot-before-color);

    animation: dot-flashing 1s infinite alternate;
    animation-delay: 0s;
  }

  &::after {
    left: $dot-spacing;

    @include dot($bg-color: $dot-after-color);

    animation: dot-flashing 1s infinite alternate;
    animation-delay: 1s;
  }
}

@keyframes dot-flashing {
  0% {
    background-color: $dot-color;
  }

  50%,
  100% {
    background-color: rgba(lighten($dot-color, 20%), 0.5);
  }
}
