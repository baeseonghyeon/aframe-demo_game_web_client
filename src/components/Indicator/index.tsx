import React from 'react';
import styles from './Indicator.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

const Indicator = () => {
  return (
    <svg className={cn('indicator')} viewBox="0 0 100 100">
      <circle className={cn('circle')} cx="50" cy="50" r="45"></circle>
    </svg>
  );
};

export default Indicator;
