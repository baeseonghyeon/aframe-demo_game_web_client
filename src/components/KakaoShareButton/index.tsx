import React, { useEffect } from 'react';
import styles from './KakaoShareButton.module.scss';
import cb from 'classnames/bind';
import kakaoIcon from './assets/kakao.png';

const cn = cb.bind(styles);

interface KakaoShareButtonProps {
  children?: React.ReactNode;
  className?: string;
  onClick?: () => void;
}

const KakaoShareButton = (props: KakaoShareButtonProps) => {
  const { children, className, onClick } = props;

  useEffect(() => {
    createKakaoButton();
  }, []);

  const createKakaoButton = () => {
    if (window.Kakao) {
      const kakao = window.Kakao;
      // 중복 initialization 방지
      if (!kakao.isInitialized()) {
        kakao.init(process.env.REACT_APP_KAKAO_KEY);
      }

      kakao.Link.createCustomButton({
        container: '#kakao-link-btn',
        templateId: 56200
      });
    }
  };

  return (
    <div className={cn('container')}>
      {/* Kakao share button */}
      <button id="kakao-link-btn" className={cn('btn')}>
        <img className={cn('icon')} src={kakaoIcon} alt="kakao-share-icon" />
      </button>
    </div>
  );
};

export default KakaoShareButton;
