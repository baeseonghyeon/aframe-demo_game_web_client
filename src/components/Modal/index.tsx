import React from 'react';
import styles from './Modal.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

export interface ModalProps extends React.HtmlHTMLAttributes<HTMLDivElement> {
  /**
   * 컨테이너 클래스 네임
   */
  containerClassName?: string;
  /**
   * 모달 여부
   */
  isVisible: boolean;
  /**
   * 백드롭 여부
   */
  isBackdrop?: boolean;
  /**
   * 백드롭 스타일
   */
  backdropClassName?: string;
  /**
   * 백드롭 클릭 이벤트
   */
  onClickBackdrop?: () => void;
  /**
   * ref
   */
  ref?: any;
}

/**
 *  모달 컴포넌트 props : ({ isVisible, onClickBackdrop, backdropClassName })
 */
const Modal = (props: ModalProps) => {
  const {
    className,
    containerClassName,
    isVisible = false,
    isBackdrop = true,
    backdropClassName,
    onClickBackdrop,
    ref,
    ...rest
  } = props;

  // Visible Class
  const visibleClassNames = cn([
    containerClassName && containerClassName,
    isVisible ? `container--visible` : `container--none`,
    isVisible && isBackdrop ? `container--fill` : `container--fit`
  ]);

  // Modal Container Class
  const classNames = cn(props.className, 'wrapper');

  return (
    <div className={visibleClassNames} ref={ref} {...rest}>
      {/* Modal Content  */}
      <div className={classNames}>{isVisible && props.children}</div>
      {/* BackDrop Layout */}
      {isBackdrop && (
        <div
          className={cn('backdrop', backdropClassName)}
          onClick={onClickBackdrop}
        ></div>
      )}
    </div>
  );
};

export default Modal;
