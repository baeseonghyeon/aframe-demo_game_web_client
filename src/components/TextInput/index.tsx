import React from 'react';
import styles from './TextInput.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

export interface TextInputProps
  extends React.HtmlHTMLAttributes<HTMLDivElement | HTMLInputElement> {
  className?: string;
  placeholder?: string;
  value: string;
  autoFocus?: boolean;
  onChange?: (e: any) => void;
}

const TextInput = (props: TextInputProps) => {
  const { placeholder, value, className, autoFocus, onChange, ...rest } = props;

  return (
    <input
      className={cn('container', className)}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      autoFocus={autoFocus}
      {...rest}
    />
  );
};

export default TextInput;
