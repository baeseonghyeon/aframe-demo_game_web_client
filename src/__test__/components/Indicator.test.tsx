import React from 'react';
import ReactDOM from 'react-dom';
import { default as Indicator } from '../../components/Indicator';
import { render } from '@testing-library/react';

describe('[Component] Indicator', () => {
  it('스냅샷 테스트', () => {
    const rendered = render(<Indicator />);
    expect(rendered.container).toMatchSnapshot();
  });
});
