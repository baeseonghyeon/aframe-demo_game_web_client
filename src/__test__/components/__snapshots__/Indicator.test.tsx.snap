// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Indicator 스냅샷 테스트 1`] = `
<div>
  <svg
    class="indicator"
    viewBox="0 0 100 100"
  >
    <circle
      class="circle"
      cx="50"
      cy="50"
      r="45"
    />
  </svg>
</div>
`;
